### PR TITLE
plugins/rest: Add option to specify CA for remote services

### PR DESCRIPTION
**Pull request by ashutosh-narkar**
This change allows users to specify a certificate for the services
that implement the bundle, status etc. APIs. This cert will be
used to create the root CA pool.

Fixes: #1954

Signed-off-by: Ashutosh Narkar <anarkar4387@gmail.com>

### CODE REVIEW

File: docs/content/configuration.md [hunk 1]
```diff
@@ -242,6 +242,8 @@ multiple services.
 | `services[_].url` | `string` | Yes | Base URL to contact the service with. |
 | `services[_].response_header_timeout_seconds` | `int64` | No (default: 10) | Amount of time to wait for a server's response headers after fully writing the request. This time does not include the time to read the response body. |
 | `services[_].headers` | `object` | No | HTTP headers to include in requests to the service. |
+| `services[_].tls.ca_cert` | `string` | No | The path to the root CA certificate. |
```
Comment (srenatus) [inline docs/content/configuration.md:245]: 💭 If these are not required, what's the default?

File: plugins/rest/rest_auth.go [hunk 2]
```diff
@@ -356,8 +385,8 @@ type clientTLSAuthPlugin struct {
 	Cert                 string `json:"cert"`
 	PrivateKey           string `json:"private_key"`
 	PrivateKeyPassphrase string `json:"private_key_passphrase,omitempty"`
-	CACert               string `json:"ca_cert,omitempty"`
-	SystemCARequired     bool   `json:"system_ca_required,omitempty"`
+	CACert               string `json:"ca_cert,omitempty"`            // Deprecated
+	SystemCARequired     bool   `json:"system_ca_required,omitempty"` // Deprecated
```
Comment (srenatus) [inline plugins/rest/rest_auth.go:389]: Will these still work? Should we log something to inform users that have configured these fields that they're deprecated...?

**Review by srenatus — COMMENTED** [review]
Looks good, two questions 🙃 👇

File: docs/content/configuration.md (hunk 1 above)
Comment (ashutosh-narkar) [reply docs/content/configuration.md]: Updated the field description to include defaults.

File: plugins/rest/rest_auth.go (hunk 2 above)
Comment (ashutosh-narkar) [reply plugins/rest/rest_auth.go]: These should work if the users haven't specified the new fields in the `services` config. Also added a deprecation log message.

**Review by srenatus — APPROVED** [review]
Thank you

**Pull request merged**